CMakeLists.txt

```diff
@@ -38,29 +42,31 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-# Enable all the warnings we can. We need noncall exceptions so we can throw exceptions from signal handlers. This
-# allows us to catch null pointer exceptions. no-int-in-bool-context is to silence a warning from Eigen.
-add_compile_options(-Wall -Wpedantic -Wextra -Wno-int-in-bool-context -fnon-call-exceptions)
-
-# C++17 allows eigen to not need to worry about alignment
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-
-# Make sure our binaries can find our libraries both in docker and on the robot
-list(APPEND CMAKE_INSTALL_RPATH /home/nubots/toolchain/)
-list(APPEND CMAKE_INSTALL_RPATH /usr/local/lib)
-
-# Disable armadillo bounds checking in release and min size release builds
-list(APPEND CMAKE_CXX_FLAGS_RELEASE "-DARMA_NO_DEBUG")
-list(APPEND CMAKE_CXX_FLAGS_MINSIZEREL "-DARMA_NO_DEBUG")
```

Comment:
why are you deleting these lines?

Reply:
Because they were solved by the modern cmake stuff and just were missed. We are setting public attributes on things we're building that require cxx_17 rather than setting them globally like this.